### PR TITLE
ref: version refactored to action pattern

### PR DESCRIPTION
**Pull request by technosophos**
**What this PR does / why we need it**:

This migrates the `helm version` logic to `pkg/action`, and also uses @adamreese 's suggested pattern of function arguments for formatting version.

**Special notes for your reviewer**:

Depends on #5077

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/action/action.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+// Timestamper is a function that can provide a timestamp.
+//
+// If this is not set, the `time.Now()` function is used to generate
+// timestamps. This may be overridden for testing.
+type Timestamper func() time.Time
+
+// Configuration injects the dependencies that all actions share.
+type Configuration struct {
+	//engine    Engine
```
Comment (adamreese) [inline pkg/action/action.go]: delete me

**Comment by adamreese** [comment]
The purpose of package `action` is to allow for helm logic to be imported into an external project.  The version command should not be included because is dependent on compile time data.  Consumers would need a way of manually setting the build info which defeats the purpose.

One possible solution... set a specific version statically and checkin to vcs.

**Comment by technosophos** [comment]
That is a very good point.

Long-term, it does seem that we'd want version info available to people using Helm as a library. So a checked-in representation of that would be good.

Would you rather that we tried to do that in this PR? Otherwise, I can just close this PR and we can deal with versioning stuff at a more opportune time.

**Comment by adamreese** [comment]
I just spiked out splitting `pkg/version` into `pkg/version` and `internal/version` . It might be better to start there

**Pull request closed without merging**